src/app/model/Productmodel.ts

```diff
@@ -39,30 +39,29 @@ export type Most = {
 };
 
 export class Productmodel {
-  counts: IProductSoldCounts;
+  public counts:any = new IProductSoldCounts();
```

Comment:
Also the naming scheme indicates that IProductSoldCounts is an interface but turns out it is a class